// @aml-org/amf-antlr-parsers/lib/internal/graphqlcb.m.js
import antlr4 from 'antlr4';
import GraphqlLexer from './graphql/GraphQLLexer.js';
import GraphqlParser from './graphql/GraphQLParser.js';
import GraphqlListener from './graphql/GraphQLListener.js';

export class GraphqlASTBuilderCB extends GraphqlListener {

    visitTerminal(node) {
        this.cb.visitTerminal(node, this.parser)
    }

    visitErrorNode(node) {
        this.cb.visitErrorNode(node, this.parser)
    }

    enterEveryRule(node) {
        this.cb.enterEveryRule(node, this.parser)
    }

    exitEveryRule(node) {
        this.cb.exitEveryRule()
    }

    run(input, cb, errorListener) {
        this.cb = cb;
        const chars = new antlr4.InputStream(input);
        const lexer = new GraphqlLexer(chars);
        lexer.removeErrorListeners();
        lexer.addErrorListener(errorListener);
        const tokens = new antlr4.CommonTokenStream(lexer);
        const parser = new GraphqlParser(tokens);
        parser.removeErrorListeners();
        parser.addErrorListener(errorListener);
        this.parser = parser
        this.parser.buildParseTrees = true;
        const tree = this.parser.document();
        antlr4.tree.ParseTreeWalker.DEFAULT.walk(this, tree);
    }
}
