// @aml-org/amf-antlr-parsers/lib/internal/protobuf3cb.m.js
import antlr4 from 'antlr4';
import Protobuf3Lexer from './proto3/Protobuf3Lexer.js';
import Protobuf3Parser from './proto3/Protobuf3Parser.js';
import Protobuf3Listener from './proto3/Protobuf3Listener.js';

export class Protobuf3ASTBuilderCB extends Protobuf3Listener {

    visitTerminal(node) {
        this.cb.visitTerminal(node, this.parser)
    }

    visitErrorNode(node) {
        this.cb.visitErrorNode(node, this.parser)
    }

    enterEveryRule(node) {
        this.cb.enterEveryRule(node, this.parser)
    }

    exitEveryRule(node) {
        this.cb.exitEveryRule()
    }

    run(input, cb) {
        this.cb = cb;
        const chars = new antlr4.InputStream(input);
        const lexer = new Protobuf3Lexer(chars);
        const tokens = new antlr4.CommonTokenStream(lexer);
        this.parser = new Protobuf3Parser(tokens);
        this.parser.buildParseTrees = true;
        const tree = this.parser.proto();
        antlr4.tree.ParseTreeWalker.DEFAULT.walk(this, tree);
    }
}
