// @aml-org/amf-antlr-parsers/lib/internal/graphqlfederation/GraphQLFederationListener.js
"use strict";

function _typeof(obj) {
  "@babel/helpers - typeof";

  if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") {
    _typeof = function _typeof(obj) {
      return typeof obj;
    };
  } else {
    _typeof = function _typeof(obj) {
      return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj;
    };
  }

  return _typeof(obj);
}

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _antlr = _interopRequireDefault(require("antlr4"));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : {
    "default": obj
  };
}

function _classCallCheck(instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError("Cannot call a class as a function");
  }
}

function _defineProperties(target, props) {
  for (var i = 0; i < props.length; i++) {
    var descriptor = props[i];
    descriptor.enumerable = descriptor.enumerable || false;
    descriptor.configurable = true;
    if ("value" in descriptor) descriptor.writable = true;
    Object.defineProperty(target, descriptor.key, descriptor);
  }
}

function _createClass(Constructor, protoProps, staticProps) {
  if (protoProps) _defineProperties(Constructor.prototype, protoProps);
  if (staticProps) _defineProperties(Constructor, staticProps);
  return Constructor;
}

function _inherits(subClass, superClass) {
  if (typeof superClass !== "function" && superClass !== null) {
    throw new TypeError("Super expression must either be null or a function");
  }

  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {
      value: subClass,
      writable: true,
      configurable: true
    }
  });
  if (superClass) _setPrototypeOf(subClass, superClass);
}

function _setPrototypeOf(o, p) {
  _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) {
    o.__proto__ = p;
    return o;
  };

  return _setPrototypeOf(o, p);
}

function _createSuper(Derived) {
  var hasNativeReflectConstruct = _isNativeReflectConstruct();

  return function _createSuperInternal() {
    var Super = _getPrototypeOf(Derived),
        result;

    if (hasNativeReflectConstruct) {
      var NewTarget = _getPrototypeOf(this).constructor;

      result = Reflect.construct(Super, arguments, NewTarget);
    } else {
      result = Super.apply(this, arguments);
    }

    return _possibleConstructorReturn(this, result);
  };
}

function _possibleConstructorReturn(self, call) {
  if (call && (_typeof(call) === "object" || typeof call === "function")) {
    return call;
  }

  return _assertThisInitialized(self);
}

function _assertThisInitialized(self) {
  if (self === void 0) {
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  }

  return self;
}

function _isNativeReflectConstruct() {
  if (typeof Reflect === "undefined" || !Reflect.construct) return false;
  if (Reflect.construct.sham) return false;
  if (typeof Proxy === "function") return true;

  try {
    Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {}));
    return true;
  } catch (e) {
    return false;
  }
}

function _getPrototypeOf(o) {
  _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) {
    return o.__proto__ || Object.getPrototypeOf(o);
  };
  return _getPrototypeOf(o);
} // This class defines a complete listener for a parse tree produced by GraphQLFederationParser.


var GraphQLFederationListener = /*#__PURE__*/function (_antlr4$tree$ParseTre) {
  _inherits(GraphQLFederationListener, _antlr4$tree$ParseTre);

  var _super = _createSuper(GraphQLFederationListener);

  function GraphQLFederationListener() {
    _classCallCheck(this, GraphQLFederationListener);

    return _super.apply(this, arguments);
  }

  _createClass(GraphQLFederationListener, [{
    key: "enterDocument",
    value: // Enter a parse tree produced by GraphQLFederationParser#document.
    function enterDocument(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#document.

  }, {
    key: "exitDocument",
    value: function exitDocument(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#definition.

  }, {
    key: "enterDefinition",
    value: function enterDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#definition.

  }, {
    key: "exitDefinition",
    value: function exitDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#executableDefinition.

  }, {
    key: "enterExecutableDefinition",
    value: function enterExecutableDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#executableDefinition.

  }, {
    key: "exitExecutableDefinition",
    value: function exitExecutableDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#operationDefinition.

  }, {
    key: "enterOperationDefinition",
    value: function enterOperationDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#operationDefinition.

  }, {
    key: "exitOperationDefinition",
    value: function exitOperationDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#operationType.

  }, {
    key: "enterOperationType",
    value: function enterOperationType(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#operationType.

  }, {
    key: "exitOperationType",
    value: function exitOperationType(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#selectionSet.

  }, {
    key: "enterSelectionSet",
    value: function enterSelectionSet(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#selectionSet.

  }, {
    key: "exitSelectionSet",
    value: function exitSelectionSet(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#selection.

  }, {
    key: "enterSelection",
    value: function enterSelection(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#selection.

  }, {
    key: "exitSelection",
    value: function exitSelection(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#field.

  }, {
    key: "enterField",
    value: function enterField(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#field.

  }, {
    key: "exitField",
    value: function exitField(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#arguments.

  }, {
    key: "enterArguments",
    value: function enterArguments(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#arguments.

  }, {
    key: "exitArguments",
    value: function exitArguments(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#argument.

  }, {
    key: "enterArgument",
    value: function enterArgument(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#argument.

  }, {
    key: "exitArgument",
    value: function exitArgument(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#alias.

  }, {
    key: "enterAlias",
    value: function enterAlias(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#alias.

  }, {
    key: "exitAlias",
    value: function exitAlias(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#fragmentSpread.

  }, {
    key: "enterFragmentSpread",
    value: function enterFragmentSpread(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#fragmentSpread.

  }, {
    key: "exitFragmentSpread",
    value: function exitFragmentSpread(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#fragmentDefinition.

  }, {
    key: "enterFragmentDefinition",
    value: function enterFragmentDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#fragmentDefinition.

  }, {
    key: "exitFragmentDefinition",
    value: function exitFragmentDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#fragmentName.

  }, {
    key: "enterFragmentName",
    value: function enterFragmentName(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#fragmentName.

  }, {
    key: "exitFragmentName",
    value: function exitFragmentName(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#typeCondition.

  }, {
    key: "enterTypeCondition",
    value: function enterTypeCondition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#typeCondition.

  }, {
    key: "exitTypeCondition",
    value: function exitTypeCondition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#inlineFragment.

  }, {
    key: "enterInlineFragment",
    value: function enterInlineFragment(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#inlineFragment.

  }, {
    key: "exitInlineFragment",
    value: function exitInlineFragment(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#value.

  }, {
    key: "enterValue",
    value: function enterValue(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#value.

  }, {
    key: "exitValue",
    value: function exitValue(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#intValue.

  }, {
    key: "enterIntValue",
    value: function enterIntValue(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#intValue.

  }, {
    key: "exitIntValue",
    value: function exitIntValue(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#floatValue.

  }, {
    key: "enterFloatValue",
    value: function enterFloatValue(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#floatValue.

  }, {
    key: "exitFloatValue",
    value: function exitFloatValue(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#booleanValue.

  }, {
    key: "enterBooleanValue",
    value: function enterBooleanValue(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#booleanValue.

  }, {
    key: "exitBooleanValue",
    value: function exitBooleanValue(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#stringValue.

  }, {
    key: "enterStringValue",
    value: function enterStringValue(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#stringValue.

  }, {
    key: "exitStringValue",
    value: function exitStringValue(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#nullValue.

  }, {
    key: "enterNullValue",
    value: function enterNullValue(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#nullValue.

  }, {
    key: "exitNullValue",
    value: function exitNullValue(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#enumValue.

  }, {
    key: "enterEnumValue",
    value: function enterEnumValue(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#enumValue.

  }, {
    key: "exitEnumValue",
    value: function exitEnumValue(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#listValue.

  }, {
    key: "enterListValue",
    value: function enterListValue(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#listValue.

  }, {
    key: "exitListValue",
    value: function exitListValue(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#objectValue.

  }, {
    key: "enterObjectValue",
    value: function enterObjectValue(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#objectValue.

  }, {
    key: "exitObjectValue",
    value: function exitObjectValue(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#objectField.

  }, {
    key: "enterObjectField",
    value: function enterObjectField(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#objectField.

  }, {
    key: "exitObjectField",
    value: function exitObjectField(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#variable.

  }, {
    key: "enterVariable",
    value: function enterVariable(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#variable.

  }, {
    key: "exitVariable",
    value: function exitVariable(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#variableDefinitions.

  }, {
    key: "enterVariableDefinitions",
    value: function enterVariableDefinitions(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#variableDefinitions.

  }, {
    key: "exitVariableDefinitions",
    value: function exitVariableDefinitions(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#variableDefinition.

  }, {
    key: "enterVariableDefinition",
    value: function enterVariableDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#variableDefinition.

  }, {
    key: "exitVariableDefinition",
    value: function exitVariableDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#defaultValue.

  }, {
    key: "enterDefaultValue",
    value: function enterDefaultValue(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#defaultValue.

  }, {
    key: "exitDefaultValue",
    value: function exitDefaultValue(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#type_.

  }, {
    key: "enterType_",
    value: function enterType_(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#type_.

  }, {
    key: "exitType_",
    value: function exitType_(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#namedType.

  }, {
    key: "enterNamedType",
    value: function enterNamedType(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#namedType.

  }, {
    key: "exitNamedType",
    value: function exitNamedType(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#listType.

  }, {
    key: "enterListType",
    value: function enterListType(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#listType.

  }, {
    key: "exitListType",
    value: function exitListType(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#directives.

  }, {
    key: "enterDirectives",
    value: function enterDirectives(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#directives.

  }, {
    key: "exitDirectives",
    value: function exitDirectives(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#directive.

  }, {
    key: "enterDirective",
    value: function enterDirective(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#directive.

  }, {
    key: "exitDirective",
    value: function exitDirective(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#typeSystemDefinition.

  }, {
    key: "enterTypeSystemDefinition",
    value: function enterTypeSystemDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#typeSystemDefinition.

  }, {
    key: "exitTypeSystemDefinition",
    value: function exitTypeSystemDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#typeSystemExtension.

  }, {
    key: "enterTypeSystemExtension",
    value: function enterTypeSystemExtension(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#typeSystemExtension.

  }, {
    key: "exitTypeSystemExtension",
    value: function exitTypeSystemExtension(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#schemaDefinition.

  }, {
    key: "enterSchemaDefinition",
    value: function enterSchemaDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#schemaDefinition.

  }, {
    key: "exitSchemaDefinition",
    value: function exitSchemaDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#rootOperationTypeDefinition.

  }, {
    key: "enterRootOperationTypeDefinition",
    value: function enterRootOperationTypeDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#rootOperationTypeDefinition.

  }, {
    key: "exitRootOperationTypeDefinition",
    value: function exitRootOperationTypeDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#schemaExtension.

  }, {
    key: "enterSchemaExtension",
    value: function enterSchemaExtension(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#schemaExtension.

  }, {
    key: "exitSchemaExtension",
    value: function exitSchemaExtension(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#operationTypeDefinition.

  }, {
    key: "enterOperationTypeDefinition",
    value: function enterOperationTypeDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#operationTypeDefinition.

  }, {
    key: "exitOperationTypeDefinition",
    value: function exitOperationTypeDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#description.

  }, {
    key: "enterDescription",
    value: function enterDescription(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#description.

  }, {
    key: "exitDescription",
    value: function exitDescription(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#typeDefinition.

  }, {
    key: "enterTypeDefinition",
    value: function enterTypeDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#typeDefinition.

  }, {
    key: "exitTypeDefinition",
    value: function exitTypeDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#typeExtension.

  }, {
    key: "enterTypeExtension",
    value: function enterTypeExtension(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#typeExtension.

  }, {
    key: "exitTypeExtension",
    value: function exitTypeExtension(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#scalarTypeDefinition.

  }, {
    key: "enterScalarTypeDefinition",
    value: function enterScalarTypeDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#scalarTypeDefinition.

  }, {
    key: "exitScalarTypeDefinition",
    value: function exitScalarTypeDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#scalarTypeExtension.

  }, {
    key: "enterScalarTypeExtension",
    value: function enterScalarTypeExtension(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#scalarTypeExtension.

  }, {
    key: "exitScalarTypeExtension",
    value: function exitScalarTypeExtension(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#objectTypeDefinition.

  }, {
    key: "enterObjectTypeDefinition",
    value: function enterObjectTypeDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#objectTypeDefinition.

  }, {
    key: "exitObjectTypeDefinition",
    value: function exitObjectTypeDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#implementsInterfaces.

  }, {
    key: "enterImplementsInterfaces",
    value: function enterImplementsInterfaces(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#implementsInterfaces.

  }, {
    key: "exitImplementsInterfaces",
    value: function exitImplementsInterfaces(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#fieldsDefinition.

  }, {
    key: "enterFieldsDefinition",
    value: function enterFieldsDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#fieldsDefinition.

  }, {
    key: "exitFieldsDefinition",
    value: function exitFieldsDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#fieldDefinition.

  }, {
    key: "enterFieldDefinition",
    value: function enterFieldDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#fieldDefinition.

  }, {
    key: "exitFieldDefinition",
    value: function exitFieldDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#argumentsDefinition.

  }, {
    key: "enterArgumentsDefinition",
    value: function enterArgumentsDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#argumentsDefinition.

  }, {
    key: "exitArgumentsDefinition",
    value: function exitArgumentsDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#inputValueDefinition.

  }, {
    key: "enterInputValueDefinition",
    value: function enterInputValueDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#inputValueDefinition.

  }, {
    key: "exitInputValueDefinition",
    value: function exitInputValueDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#objectTypeExtension.

  }, {
    key: "enterObjectTypeExtension",
    value: function enterObjectTypeExtension(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#objectTypeExtension.

  }, {
    key: "exitObjectTypeExtension",
    value: function exitObjectTypeExtension(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#interfaceTypeDefinition.

  }, {
    key: "enterInterfaceTypeDefinition",
    value: function enterInterfaceTypeDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#interfaceTypeDefinition.

  }, {
    key: "exitInterfaceTypeDefinition",
    value: function exitInterfaceTypeDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#interfaceTypeExtension.

  }, {
    key: "enterInterfaceTypeExtension",
    value: function enterInterfaceTypeExtension(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#interfaceTypeExtension.

  }, {
    key: "exitInterfaceTypeExtension",
    value: function exitInterfaceTypeExtension(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#unionTypeDefinition.

  }, {
    key: "enterUnionTypeDefinition",
    value: function enterUnionTypeDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#unionTypeDefinition.

  }, {
    key: "exitUnionTypeDefinition",
    value: function exitUnionTypeDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#unionMemberTypes.

  }, {
    key: "enterUnionMemberTypes",
    value: function enterUnionMemberTypes(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#unionMemberTypes.

  }, {
    key: "exitUnionMemberTypes",
    value: function exitUnionMemberTypes(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#unionTypeExtension.

  }, {
    key: "enterUnionTypeExtension",
    value: function enterUnionTypeExtension(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#unionTypeExtension.

  }, {
    key: "exitUnionTypeExtension",
    value: function exitUnionTypeExtension(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#enumTypeDefinition.

  }, {
    key: "enterEnumTypeDefinition",
    value: function enterEnumTypeDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#enumTypeDefinition.

  }, {
    key: "exitEnumTypeDefinition",
    value: function exitEnumTypeDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#enumValuesDefinition.

  }, {
    key: "enterEnumValuesDefinition",
    value: function enterEnumValuesDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#enumValuesDefinition.

  }, {
    key: "exitEnumValuesDefinition",
    value: function exitEnumValuesDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#enumValueDefinition.

  }, {
    key: "enterEnumValueDefinition",
    value: function enterEnumValueDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#enumValueDefinition.

  }, {
    key: "exitEnumValueDefinition",
    value: function exitEnumValueDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#enumTypeExtension.

  }, {
    key: "enterEnumTypeExtension",
    value: function enterEnumTypeExtension(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#enumTypeExtension.

  }, {
    key: "exitEnumTypeExtension",
    value: function exitEnumTypeExtension(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#inputObjectTypeDefinition.

  }, {
    key: "enterInputObjectTypeDefinition",
    value: function enterInputObjectTypeDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#inputObjectTypeDefinition.

  }, {
    key: "exitInputObjectTypeDefinition",
    value: function exitInputObjectTypeDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#inputFieldsDefinition.

  }, {
    key: "enterInputFieldsDefinition",
    value: function enterInputFieldsDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#inputFieldsDefinition.

  }, {
    key: "exitInputFieldsDefinition",
    value: function exitInputFieldsDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#inputObjectTypeExtension.

  }, {
    key: "enterInputObjectTypeExtension",
    value: function enterInputObjectTypeExtension(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#inputObjectTypeExtension.

  }, {
    key: "exitInputObjectTypeExtension",
    value: function exitInputObjectTypeExtension(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#directiveDefinition.

  }, {
    key: "enterDirectiveDefinition",
    value: function enterDirectiveDefinition(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#directiveDefinition.

  }, {
    key: "exitDirectiveDefinition",
    value: function exitDirectiveDefinition(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#directiveLocations.

  }, {
    key: "enterDirectiveLocations",
    value: function enterDirectiveLocations(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#directiveLocations.

  }, {
    key: "exitDirectiveLocations",
    value: function exitDirectiveLocations(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#directiveLocation.

  }, {
    key: "enterDirectiveLocation",
    value: function enterDirectiveLocation(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#directiveLocation.

  }, {
    key: "exitDirectiveLocation",
    value: function exitDirectiveLocation(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#executableDirectiveLocation.

  }, {
    key: "enterExecutableDirectiveLocation",
    value: function enterExecutableDirectiveLocation(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#executableDirectiveLocation.

  }, {
    key: "exitExecutableDirectiveLocation",
    value: function exitExecutableDirectiveLocation(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#typeSystemDirectiveLocation.

  }, {
    key: "enterTypeSystemDirectiveLocation",
    value: function enterTypeSystemDirectiveLocation(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#typeSystemDirectiveLocation.

  }, {
    key: "exitTypeSystemDirectiveLocation",
    value: function exitTypeSystemDirectiveLocation(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#externalDirective.

  }, {
    key: "enterExternalDirective",
    value: function enterExternalDirective(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#externalDirective.

  }, {
    key: "exitExternalDirective",
    value: function exitExternalDirective(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#requiresDirective.

  }, {
    key: "enterRequiresDirective",
    value: function enterRequiresDirective(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#requiresDirective.

  }, {
    key: "exitRequiresDirective",
    value: function exitRequiresDirective(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#providesDirective.

  }, {
    key: "enterProvidesDirective",
    value: function enterProvidesDirective(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#providesDirective.

  }, {
    key: "exitProvidesDirective",
    value: function exitProvidesDirective(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#keyDirective.

  }, {
    key: "enterKeyDirective",
    value: function enterKeyDirective(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#keyDirective.

  }, {
    key: "exitKeyDirective",
    value: function exitKeyDirective(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#shareableDirective.

  }, {
    key: "enterShareableDirective",
    value: function enterShareableDirective(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#shareableDirective.

  }, {
    key: "exitShareableDirective",
    value: function exitShareableDirective(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#overrideDirective.

  }, {
    key: "enterOverrideDirective",
    value: function enterOverrideDirective(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#overrideDirective.

  }, {
    key: "exitOverrideDirective",
    value: function exitOverrideDirective(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#fieldSet.

  }, {
    key: "enterFieldSet",
    value: function enterFieldSet(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#fieldSet.

  }, {
    key: "exitFieldSet",
    value: function exitFieldSet(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#fieldSetComponent.

  }, {
    key: "enterFieldSetComponent",
    value: function enterFieldSetComponent(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#fieldSetComponent.

  }, {
    key: "exitFieldSetComponent",
    value: function exitFieldSetComponent(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#nestedFieldSet.

  }, {
    key: "enterNestedFieldSet",
    value: function enterNestedFieldSet(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#nestedFieldSet.

  }, {
    key: "exitNestedFieldSet",
    value: function exitNestedFieldSet(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#fieldDefinitionFederationDirectives.

  }, {
    key: "enterFieldDefinitionFederationDirectives",
    value: function enterFieldDefinitionFederationDirectives(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#fieldDefinitionFederationDirectives.

  }, {
    key: "exitFieldDefinitionFederationDirectives",
    value: function exitFieldDefinitionFederationDirectives(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#typeDefinitionFederationDirectives.

  }, {
    key: "enterTypeDefinitionFederationDirectives",
    value: function enterTypeDefinitionFederationDirectives(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#typeDefinitionFederationDirectives.

  }, {
    key: "exitTypeDefinitionFederationDirectives",
    value: function exitTypeDefinitionFederationDirectives(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#typeExtensionFederationDirectives.

  }, {
    key: "enterTypeExtensionFederationDirectives",
    value: function enterTypeExtensionFederationDirectives(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#typeExtensionFederationDirectives.

  }, {
    key: "exitTypeExtensionFederationDirectives",
    value: function exitTypeExtensionFederationDirectives(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#name.

  }, {
    key: "enterName",
    value: function enterName(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#name.

  }, {
    key: "exitName",
    value: function exitName(ctx) {} // Enter a parse tree produced by GraphQLFederationParser#keyword.

  }, {
    key: "enterKeyword",
    value: function enterKeyword(ctx) {} // Exit a parse tree produced by GraphQLFederationParser#keyword.

  }, {
    key: "exitKeyword",
    value: function exitKeyword(ctx) {}
  }]);

  return GraphQLFederationListener;
}(_antlr["default"].tree.ParseTreeListener);

exports["default"] = GraphQLFederationListener;
