// @aml-org/amf-antlr-parsers/lib/internal/proto3/Protobuf3Listener.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _antlr = _interopRequireDefault(require("antlr4"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); Object.defineProperty(subClass, "prototype", { writable: false }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } else if (call !== void 0) { throw new TypeError("Derived constructors may only return object or undefined"); } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

// This class defines a complete listener for a parse tree produced by Protobuf3Parser.
var Protobuf3Listener = /*#__PURE__*/function (_antlr4$tree$ParseTre) {
  _inherits(Protobuf3Listener, _antlr4$tree$ParseTre);

  var _super = _createSuper(Protobuf3Listener);

  function Protobuf3Listener() {
    _classCallCheck(this, Protobuf3Listener);

    return _super.apply(this, arguments);
  }

  _createClass(Protobuf3Listener, [{
    key: "enterProto",
    value: // Enter a parse tree produced by Protobuf3Parser#proto.
    function enterProto(ctx) {} // Exit a parse tree produced by Protobuf3Parser#proto.

  }, {
    key: "exitProto",
    value: function exitProto(ctx) {} // Enter a parse tree produced by Protobuf3Parser#syntax.

  }, {
    key: "enterSyntax",
    value: function enterSyntax(ctx) {} // Exit a parse tree produced by Protobuf3Parser#syntax.

  }, {
    key: "exitSyntax",
    value: function exitSyntax(ctx) {} // Enter a parse tree produced by Protobuf3Parser#importStatement.

  }, {
    key: "enterImportStatement",
    value: function enterImportStatement(ctx) {} // Exit a parse tree produced by Protobuf3Parser#importStatement.

  }, {
    key: "exitImportStatement",
    value: function exitImportStatement(ctx) {} // Enter a parse tree produced by Protobuf3Parser#packageStatement.

  }, {
    key: "enterPackageStatement",
    value: function enterPackageStatement(ctx) {} // Exit a parse tree produced by Protobuf3Parser#packageStatement.

  }, {
    key: "exitPackageStatement",
    value: function exitPackageStatement(ctx) {} // Enter a parse tree produced by Protobuf3Parser#optionStatement.

  }, {
    key: "enterOptionStatement",
    value: function enterOptionStatement(ctx) {} // Exit a parse tree produced by Protobuf3Parser#optionStatement.

  }, {
    key: "exitOptionStatement",
    value: function exitOptionStatement(ctx) {} // Enter a parse tree produced by Protobuf3Parser#optionName.

  }, {
    key: "enterOptionName",
    value: function enterOptionName(ctx) {} // Exit a parse tree produced by Protobuf3Parser#optionName.

  }, {
    key: "exitOptionName",
    value: function exitOptionName(ctx) {} // Enter a parse tree produced by Protobuf3Parser#extendsStatement.

  }, {
    key: "enterExtendsStatement",
    value: function enterExtendsStatement(ctx) {} // Exit a parse tree produced by Protobuf3Parser#extendsStatement.

  }, {
    key: "exitExtendsStatement",
    value: function exitExtendsStatement(ctx) {} // Enter a parse tree produced by Protobuf3Parser#extendedIdentifier.

  }, {
    key: "enterExtendedIdentifier",
    value: function enterExtendedIdentifier(ctx) {} // Exit a parse tree produced by Protobuf3Parser#extendedIdentifier.

  }, {
    key: "exitExtendedIdentifier",
    value: function exitExtendedIdentifier(ctx) {} // Enter a parse tree produced by Protobuf3Parser#field.

  }, {
    key: "enterField",
    value: function enterField(ctx) {} // Exit a parse tree produced by Protobuf3Parser#field.

  }, {
    key: "exitField",
    value: function exitField(ctx) {} // Enter a parse tree produced by Protobuf3Parser#fieldOptions.

  }, {
    key: "enterFieldOptions",
    value: function enterFieldOptions(ctx) {} // Exit a parse tree produced by Protobuf3Parser#fieldOptions.

  }, {
    key: "exitFieldOptions",
    value: function exitFieldOptions(ctx) {} // Enter a parse tree produced by Protobuf3Parser#fieldOption.

  }, {
    key: "enterFieldOption",
    value: function enterFieldOption(ctx) {} // Exit a parse tree produced by Protobuf3Parser#fieldOption.

  }, {
    key: "exitFieldOption",
    value: function exitFieldOption(ctx) {} // Enter a parse tree produced by Protobuf3Parser#fieldNumber.

  }, {
    key: "enterFieldNumber",
    value: function enterFieldNumber(ctx) {} // Exit a parse tree produced by Protobuf3Parser#fieldNumber.

  }, {
    key: "exitFieldNumber",
    value: function exitFieldNumber(ctx) {} // Enter a parse tree produced by Protobuf3Parser#oneof.

  }, {
    key: "enterOneof",
    value: function enterOneof(ctx) {} // Exit a parse tree produced by Protobuf3Parser#oneof.

  }, {
    key: "exitOneof",
    value: function exitOneof(ctx) {} // Enter a parse tree produced by Protobuf3Parser#oneofField.

  }, {
    key: "enterOneofField",
    value: function enterOneofField(ctx) {} // Exit a parse tree produced by Protobuf3Parser#oneofField.

  }, {
    key: "exitOneofField",
    value: function exitOneofField(ctx) {} // Enter a parse tree produced by Protobuf3Parser#mapField.

  }, {
    key: "enterMapField",
    value: function enterMapField(ctx) {} // Exit a parse tree produced by Protobuf3Parser#mapField.

  }, {
    key: "exitMapField",
    value: function exitMapField(ctx) {} // Enter a parse tree produced by Protobuf3Parser#keyType.

  }, {
    key: "enterKeyType",
    value: function enterKeyType(ctx) {} // Exit a parse tree produced by Protobuf3Parser#keyType.

  }, {
    key: "exitKeyType",
    value: function exitKeyType(ctx) {} // Enter a parse tree produced by Protobuf3Parser#type_.

  }, {
    key: "enterType_",
    value: function enterType_(ctx) {} // Exit a parse tree produced by Protobuf3Parser#type_.

  }, {
    key: "exitType_",
    value: function exitType_(ctx) {} // Enter a parse tree produced by Protobuf3Parser#reserved.

  }, {
    key: "enterReserved",
    value: function enterReserved(ctx) {} // Exit a parse tree produced by Protobuf3Parser#reserved.

  }, {
    key: "exitReserved",
    value: function exitReserved(ctx) {} // Enter a parse tree produced by Protobuf3Parser#ranges.

  }, {
    key: "enterRanges",
    value: function enterRanges(ctx) {} // Exit a parse tree produced by Protobuf3Parser#ranges.

  }, {
    key: "exitRanges",
    value: function exitRanges(ctx) {} // Enter a parse tree produced by Protobuf3Parser#range_.

  }, {
    key: "enterRange_",
    value: function enterRange_(ctx) {} // Exit a parse tree produced by Protobuf3Parser#range_.

  }, {
    key: "exitRange_",
    value: function exitRange_(ctx) {} // Enter a parse tree produced by Protobuf3Parser#reservedFieldNames.

  }, {
    key: "enterReservedFieldNames",
    value: function enterReservedFieldNames(ctx) {} // Exit a parse tree produced by Protobuf3Parser#reservedFieldNames.

  }, {
    key: "exitReservedFieldNames",
    value: function exitReservedFieldNames(ctx) {} // Enter a parse tree produced by Protobuf3Parser#topLevelDef.

  }, {
    key: "enterTopLevelDef",
    value: function enterTopLevelDef(ctx) {} // Exit a parse tree produced by Protobuf3Parser#topLevelDef.

  }, {
    key: "exitTopLevelDef",
    value: function exitTopLevelDef(ctx) {} // Enter a parse tree produced by Protobuf3Parser#enumDef.

  }, {
    key: "enterEnumDef",
    value: function enterEnumDef(ctx) {} // Exit a parse tree produced by Protobuf3Parser#enumDef.

  }, {
    key: "exitEnumDef",
    value: function exitEnumDef(ctx) {} // Enter a parse tree produced by Protobuf3Parser#enumBody.

  }, {
    key: "enterEnumBody",
    value: function enterEnumBody(ctx) {} // Exit a parse tree produced by Protobuf3Parser#enumBody.

  }, {
    key: "exitEnumBody",
    value: function exitEnumBody(ctx) {} // Enter a parse tree produced by Protobuf3Parser#enumElement.

  }, {
    key: "enterEnumElement",
    value: function enterEnumElement(ctx) {} // Exit a parse tree produced by Protobuf3Parser#enumElement.

  }, {
    key: "exitEnumElement",
    value: function exitEnumElement(ctx) {} // Enter a parse tree produced by Protobuf3Parser#enumField.

  }, {
    key: "enterEnumField",
    value: function enterEnumField(ctx) {} // Exit a parse tree produced by Protobuf3Parser#enumField.

  }, {
    key: "exitEnumField",
    value: function exitEnumField(ctx) {} // Enter a parse tree produced by Protobuf3Parser#enumValueOptions.

  }, {
    key: "enterEnumValueOptions",
    value: function enterEnumValueOptions(ctx) {} // Exit a parse tree produced by Protobuf3Parser#enumValueOptions.

  }, {
    key: "exitEnumValueOptions",
    value: function exitEnumValueOptions(ctx) {} // Enter a parse tree produced by Protobuf3Parser#enumValueOption.

  }, {
    key: "enterEnumValueOption",
    value: function enterEnumValueOption(ctx) {} // Exit a parse tree produced by Protobuf3Parser#enumValueOption.

  }, {
    key: "exitEnumValueOption",
    value: function exitEnumValueOption(ctx) {} // Enter a parse tree produced by Protobuf3Parser#messageDef.

  }, {
    key: "enterMessageDef",
    value: function enterMessageDef(ctx) {} // Exit a parse tree produced by Protobuf3Parser#messageDef.

  }, {
    key: "exitMessageDef",
    value: function exitMessageDef(ctx) {} // Enter a parse tree produced by Protobuf3Parser#messageBody.

  }, {
    key: "enterMessageBody",
    value: function enterMessageBody(ctx) {} // Exit a parse tree produced by Protobuf3Parser#messageBody.

  }, {
    key: "exitMessageBody",
    value: function exitMessageBody(ctx) {} // Enter a parse tree produced by Protobuf3Parser#messageElement.

  }, {
    key: "enterMessageElement",
    value: function enterMessageElement(ctx) {} // Exit a parse tree produced by Protobuf3Parser#messageElement.

  }, {
    key: "exitMessageElement",
    value: function exitMessageElement(ctx) {} // Enter a parse tree produced by Protobuf3Parser#serviceDef.

  }, {
    key: "enterServiceDef",
    value: function enterServiceDef(ctx) {} // Exit a parse tree produced by Protobuf3Parser#serviceDef.

  }, {
    key: "exitServiceDef",
    value: function exitServiceDef(ctx) {} // Enter a parse tree produced by Protobuf3Parser#serviceElement.

  }, {
    key: "enterServiceElement",
    value: function enterServiceElement(ctx) {} // Exit a parse tree produced by Protobuf3Parser#serviceElement.

  }, {
    key: "exitServiceElement",
    value: function exitServiceElement(ctx) {} // Enter a parse tree produced by Protobuf3Parser#rpc.

  }, {
    key: "enterRpc",
    value: function enterRpc(ctx) {} // Exit a parse tree produced by Protobuf3Parser#rpc.

  }, {
    key: "exitRpc",
    value: function exitRpc(ctx) {} // Enter a parse tree produced by Protobuf3Parser#constant.

  }, {
    key: "enterConstant",
    value: function enterConstant(ctx) {} // Exit a parse tree produced by Protobuf3Parser#constant.

  }, {
    key: "exitConstant",
    value: function exitConstant(ctx) {} // Enter a parse tree produced by Protobuf3Parser#blockLit.

  }, {
    key: "enterBlockLit",
    value: function enterBlockLit(ctx) {} // Exit a parse tree produced by Protobuf3Parser#blockLit.

  }, {
    key: "exitBlockLit",
    value: function exitBlockLit(ctx) {} // Enter a parse tree produced by Protobuf3Parser#emptyStatement.

  }, {
    key: "enterEmptyStatement",
    value: function enterEmptyStatement(ctx) {} // Exit a parse tree produced by Protobuf3Parser#emptyStatement.

  }, {
    key: "exitEmptyStatement",
    value: function exitEmptyStatement(ctx) {} // Enter a parse tree produced by Protobuf3Parser#ident.

  }, {
    key: "enterIdent",
    value: function enterIdent(ctx) {} // Exit a parse tree produced by Protobuf3Parser#ident.

  }, {
    key: "exitIdent",
    value: function exitIdent(ctx) {} // Enter a parse tree produced by Protobuf3Parser#fullIdent.

  }, {
    key: "enterFullIdent",
    value: function enterFullIdent(ctx) {} // Exit a parse tree produced by Protobuf3Parser#fullIdent.

  }, {
    key: "exitFullIdent",
    value: function exitFullIdent(ctx) {} // Enter a parse tree produced by Protobuf3Parser#messageName.

  }, {
    key: "enterMessageName",
    value: function enterMessageName(ctx) {} // Exit a parse tree produced by Protobuf3Parser#messageName.

  }, {
    key: "exitMessageName",
    value: function exitMessageName(ctx) {} // Enter a parse tree produced by Protobuf3Parser#enumName.

  }, {
    key: "enterEnumName",
    value: function enterEnumName(ctx) {} // Exit a parse tree produced by Protobuf3Parser#enumName.

  }, {
    key: "exitEnumName",
    value: function exitEnumName(ctx) {} // Enter a parse tree produced by Protobuf3Parser#fieldName.

  }, {
    key: "enterFieldName",
    value: function enterFieldName(ctx) {} // Exit a parse tree produced by Protobuf3Parser#fieldName.

  }, {
    key: "exitFieldName",
    value: function exitFieldName(ctx) {} // Enter a parse tree produced by Protobuf3Parser#oneofName.

  }, {
    key: "enterOneofName",
    value: function enterOneofName(ctx) {} // Exit a parse tree produced by Protobuf3Parser#oneofName.

  }, {
    key: "exitOneofName",
    value: function exitOneofName(ctx) {} // Enter a parse tree produced by Protobuf3Parser#mapName.

  }, {
    key: "enterMapName",
    value: function enterMapName(ctx) {} // Exit a parse tree produced by Protobuf3Parser#mapName.

  }, {
    key: "exitMapName",
    value: function exitMapName(ctx) {} // Enter a parse tree produced by Protobuf3Parser#serviceName.

  }, {
    key: "enterServiceName",
    value: function enterServiceName(ctx) {} // Exit a parse tree produced by Protobuf3Parser#serviceName.

  }, {
    key: "exitServiceName",
    value: function exitServiceName(ctx) {} // Enter a parse tree produced by Protobuf3Parser#rpcName.

  }, {
    key: "enterRpcName",
    value: function enterRpcName(ctx) {} // Exit a parse tree produced by Protobuf3Parser#rpcName.

  }, {
    key: "exitRpcName",
    value: function exitRpcName(ctx) {} // Enter a parse tree produced by Protobuf3Parser#messageType.

  }, {
    key: "enterMessageType",
    value: function enterMessageType(ctx) {} // Exit a parse tree produced by Protobuf3Parser#messageType.

  }, {
    key: "exitMessageType",
    value: function exitMessageType(ctx) {} // Enter a parse tree produced by Protobuf3Parser#enumType.

  }, {
    key: "enterEnumType",
    value: function enterEnumType(ctx) {} // Exit a parse tree produced by Protobuf3Parser#enumType.

  }, {
    key: "exitEnumType",
    value: function exitEnumType(ctx) {} // Enter a parse tree produced by Protobuf3Parser#intLit.

  }, {
    key: "enterIntLit",
    value: function enterIntLit(ctx) {} // Exit a parse tree produced by Protobuf3Parser#intLit.

  }, {
    key: "exitIntLit",
    value: function exitIntLit(ctx) {} // Enter a parse tree produced by Protobuf3Parser#strLit.

  }, {
    key: "enterStrLit",
    value: function enterStrLit(ctx) {} // Exit a parse tree produced by Protobuf3Parser#strLit.

  }, {
    key: "exitStrLit",
    value: function exitStrLit(ctx) {} // Enter a parse tree produced by Protobuf3Parser#boolLit.

  }, {
    key: "enterBoolLit",
    value: function enterBoolLit(ctx) {} // Exit a parse tree produced by Protobuf3Parser#boolLit.

  }, {
    key: "exitBoolLit",
    value: function exitBoolLit(ctx) {} // Enter a parse tree produced by Protobuf3Parser#floatLit.

  }, {
    key: "enterFloatLit",
    value: function enterFloatLit(ctx) {} // Exit a parse tree produced by Protobuf3Parser#floatLit.

  }, {
    key: "exitFloatLit",
    value: function exitFloatLit(ctx) {} // Enter a parse tree produced by Protobuf3Parser#keywords.

  }, {
    key: "enterKeywords",
    value: function enterKeywords(ctx) {} // Exit a parse tree produced by Protobuf3Parser#keywords.

  }, {
    key: "exitKeywords",
    value: function exitKeywords(ctx) {}
  }]);

  return Protobuf3Listener;
}(_antlr["default"].tree.ParseTreeListener);

exports["default"] = Protobuf3Listener;
